src/binder/bind_node_visitor.cpp

```diff
@@ -18,20 +18,28 @@
 namespace peloton {
 namespace binder {
 
-BindNodeVisitor::BindNodeVisitor(concurrency::Transaction *txn) : txn_(txn) {
+BindNodeVisitor::BindNodeVisitor(
+  concurrency::Transaction *txn, 
+  std::string default_database_name) 
+: txn_(txn),
+  default_database_name_(default_database_name) {
   context_ = nullptr;
 }
 
 void BindNodeVisitor::BindNameToNode(parser::SQLStatement *tree) {
   tree->Accept(this);
 }
 
-void BindNodeVisitor::Visit(const parser::SelectStatement *node) {
+void BindNodeVisitor::Visit(parser::SelectStatement *node) {
   // Save the upper level context
   auto pre_context = context_;
   context_ = std::make_shared<BinderContext>();
   context_->upper_context = pre_context;
-  if (node->from_table != nullptr) node->from_table->Accept(this);
+  if (node->from_table != nullptr){
+    node->from_table->Accept(this);
+    node->from_table->TryBindDatabaseName(default_database_name_);
```

Comment:
One minor thing: is this necessary if we already call TryBindDatabaseName in context_->AddTable()?

Reply:
I tested it and it seems that this line is not necessary.
One thing I do not understand is that the from_table is Accepting 'this', which is the current node instead of the from_table itself. How could this bind the from_table?